### PR TITLE
upgrade to lucene 6.0.0-snapshot-bea235f

### PR DESCRIPTION
**Pull request by rmuir**
A minimal upgrade to lucene 6 (though with minor cleanups to not leave a COMPLETE mess). 

It uses nothing new at all, cleanups etc are intentionally kept to be relatively minimal. lucene major version upgrades are chaotic enough without that.

Tests pass, it would be good to have in jenkins to help test the upcoming lucene release.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/Version.java [hunk 1]
```diff
@@ -117,130 +118,130 @@
...
+    public static final Version V_1_7_2 = new Version(V_1_7_2_ID, LUCENE_3_EMULATION_VERSION);
     public static final int V_1_7_3_ID = 1070399;
-    public static final Version V_1_7_3 = new Version(V_1_7_3_ID, org.apache.lucene.util.Version.LUCENE_4_10_4);
+    public static final Version V_1_7_3 = new Version(V_1_7_3_ID, LUCENE_3_EMULATION_VERSION);
     public static final int V_1_7_4_ID = 1070499;
-    public static final Version V_1_7_4 = new Version(V_1_7_4_ID, org.apache.lucene.util.Version.LUCENE_4_10_4);
+    public static final Version V_1_7_4 = new Version(V_1_7_4_ID, LUCENE_3_EMULATION_VERSION);
     public static final int V_1_7_5_ID = 1070599;
-    public static final Version V_1_7_5 = new Version(V_1_7_5_ID, org.apache.lucene.util.Version.LUCENE_4_10_4);
+    public static final Version V_1_7_5 = new Version(V_1_7_5_ID, LUCENE_3_EMULATION_VERSION);
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/Version.java]: I will open an issue to get rid of those ancient version... we shouldn't need this emulation version

File: core/src/main/java/org/elasticsearch/common/lucene/Lucene.java [hunk 2]
```diff
@@ -236,13 +236,8 @@ public static void checkSegmentInfoIntegrity(final Directory directory) throws I
             protected Object doBody(String segmentFileName) throws IOException {
                 try (IndexInput input = directory.openInput(segmentFileName, IOContext.READ)) {
                     final int format = input.readInt();
-                    final int actualFormat;
                     if (format == CodecUtil.CODEC_MAGIC) {
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/common/lucene/Lucene.java]: can we drop that  `if (format == CodecUtil.CODEC_MAGIC) {` here as well I mean we can't even pretend we checked it no?

Comment (rmuir) [inline core/src/main/java/org/elasticsearch/common/lucene/Lucene.java]: I saw the leniency too. but can we break this out? I don't think these kinds of changes belong in a lucene upgrade!

Comment (s1monw) [inline core/src/main/java/org/elasticsearch/common/lucene/Lucene.java]: sure I just wanted to make sure it wasn't missed

**Comment by s1monw** [comment]
LGTM

**Comment by jpountz** [comment]
LGTM

**Comment by s1monw** [comment]
I opened [this](https://github.com/elastic/elasticsearch/pull/16982/) as a followup

**Pull request closed without merging**